crates/codegen/runtime/cargo/crate/src/runtime/parser/parser_support/parser_function.rs

```diff
@@ -85,14 +87,9 @@ where
                     TerminalKind::UNRECOGNIZED
                 };
                 let node = Node::terminal(kind, input.to_string());
-                let tree = if no_match.kind.is_none() || start.utf8 == 0 {
-                    node
-                } else {
-                    trivia_nodes.push(Edge::anonymous(node));
-                    Node::nonterminal(no_match.kind.unwrap(), trivia_nodes)
-                };
+                trivia_nodes.push(Edge::anonymous(node));
                 ParseOutput {
-                    tree,
+                    tree: Rc::new(NonterminalNode::new(expected, trivia_nodes)),
```

Comment:
nit suggestion: renaming `trivia_nodes` to `children`, since it is no longer just the trivia:

```suggestion
                    tree: Rc::new(NonterminalNode::new(expected, children)),
```